framework/math/quadratures/angular/triangle_quadrature.cc

```diff
@@ -0,0 +1,476 @@
+
+  for (size_t pos = 0; pos < GAUSS_LEG_QUAD.qpoints.size(); ++pos)
+  {
+    if (GAUSS_LEG_QUAD.qpoints[pos].x < 0)
+      continue;
+    newZi.push_back(GAUSS_LEG_QUAD.qpoints[pos].x);
+    newWeights.push_back(GAUSS_LEG_QUAD.weights[pos]);
+  }
+
+  int num_div = 1;
+  int weightPos = 0;
+  for (const auto& QUAD_POINT : GAUSS_LEG_QUAD.qpoints)
+  {
+    double deltaVPhi = M_PI / (2.0 * static_cast<double>(num_div));
+    // When the GaussLegendre quadrature gives us the x points we will
+    // use for our z positions on the unit sphere, they are ordered in
+    // descending order - largest magnitude towards 0 on the negative side,
+    // and ascending order on the positive side of 0
+    // The weights_ are defined using the weights_ given by the GL quadrature
+    // which is the position in the weights_ that weightPos keeps track of.
+    // This will ignore the positive x values, and just use the descending
+    // order given by the quadrature and use the absolute value of the x values.
+
+    if (QUAD_POINT.x >= 0.0)
+      break;
+
+    for (int v = 0; v < num_div; ++v)
+    {
+      double new_z_value = abs(QUAD_POINT.x);
+      double phi = deltaVPhi / 2.0 + (double)v * deltaVPhi;
+      double theta = acos(new_z_value);
+      double sinTheta = sqrt(1 - new_z_value * new_z_value);
+      //      double weightCurrent = GAUSS_LEG_QUAD.weights[weightPos] / (num_div);
+
+      new_omega.x = sinTheta * cos(phi);
+      new_omega.y = sinTheta * sin(phi);
+      new_omega.z = new_z_value;
+      weights_.push_back(GAUSS_LEG_QUAD.weights[weightPos] * (M_PI / (2.0 * num_div)));
+      omegas_.emplace_back(new_omega);
+      abscissae_.emplace_back(phi, theta);
+    }
+
+    weightPos++;
+    num_div++;
+  }
+
+  // This is the number of points in 1 octant
+  size_t octSize = weights_.size();
+  for (int octant = 1; octant <= 3; ++octant)
+  {
+    // This is how much should be added to each phi
+    // to get the orientation right off the first index
+    double offset = M_PI_2 * octant;
+    for (size_t point = 0; point < octSize; ++point)
+    {
+      double phi = abscissae_[point].phi + offset;
+      double theta = abscissae_[point].theta;
+      double new_z_value = omegas_[point].z;
+      double sinTheta = sqrt(1 - new_z_value * new_z_value);
+
+      new_omega.x = sinTheta * cos(phi); // omegas_[l].x*xsign;
+      new_omega.y = sinTheta * sin(phi); // omegas_[l].y*ysign;
+      new_omega.z = omegas_[point].z;
+      weights_.push_back(weights_[point]);
+      omegas_.emplace_back(new_omega);
+      abscissae_.emplace_back(phi, theta);
+    }
+  }
+
+  // This builds the negative zi values
+  size_t hemisize = weights_.size();
+  for (size_t point = 0; point < hemisize; ++point)
+  {
+    new_omega.x = omegas_[point].x; // omegas[l].x*xsign;
+    new_omega.y = omegas_[point].y; // omegas[l].y*ysign;
+    new_omega.z = -omegas_[point].z;
+    double new_z_value = -omegas_[point].z;
```

Comment:
Instead of declaring these new variables, you can use the expressions inline in the lines below.